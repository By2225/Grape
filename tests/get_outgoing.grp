fun Int main(){
    Node<Int> a = '5';
    Graph<Int, Int> graph = << a -6- '7' >>;
    List<Edge<Int> > nodeList = graph.outgoing(a);
    Edge<Int> edge= nodeList[0];
    Node<Int> toNode = edge.to;
    print(toNode.val);
    return 0;
}
