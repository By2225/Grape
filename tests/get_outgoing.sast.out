
fun Int main() {
Node<Int> (Node<Int> : a = (Node<Int> : '(Int : 5)'));
Graph<Node<Int>, Edge<Int->Int>> (Graph<Node<Int>, Edge<Int->Int>> : graph = (Graph<Node<Int>, Edge<Int->*>> : <(Node<Int> : a) (Edge<Int->*> : -(Int : 6)-) (Node<Int> : '(Int : 7)') (Void : )>));
List<Edge<Int->*>> (List<Edge<Int->Int>> : nodeList = (List<Edge<Int->Int>> : (Graph<Node<Int>, Edge<Int->Int>> : graph)outgoing((Node<Int> : a))));
Edge<Int->*> (Edge<Int->*> : edge = (Edge<Int->*> : (List<Edge<Int->*>> : nodeList)[(Int : 0)]));
Node<Int> (Node<*> : toNode = (Node<*> : (Edge<Int->*> : edge).to));
(* : print((Int : (Node<Int> : toNode).val)));
return (Int : 0);
}
